Vagrantfile

```diff
@@ -88,37 +66,49 @@ Vagrant.configure('2') do |config|
         config.vm.synced_folder local_site_path(site), nfs_path(name), type: 'nfs'
         config.bindfs.bind_folder nfs_path(name), remote_site_path(name, site), u: 'vagrant', g: 'www-data', o: 'nonempty'
       end
+
       config.vm.synced_folder ANSIBLE_PATH, '/ansible-nfs', type: 'nfs'
       config.bindfs.bind_folder '/ansible-nfs', ANSIBLE_PATH_ON_VM, o: 'nonempty', p: '0644,a+D'
       config.bindfs.bind_folder bin_path, bin_path, perms: '0755'
     end
   end
 
+  vconfig.fetch('vagrant_synced_folders', []).each do |folder|
+    options = {
+      type: folder.fetch('type', 'nfs'),
+      create: folder.fetch('create', false),
+      mount_options: folder.fetch('mount_options', [])
+    }
+
+    config.vm.synced_folder folder['local_path'], folder['destination'], options
+
+    if folder.fetch('bindfs', true)
+      config.bindfs.bind_folder folder['local_path'], folder['destination'], options
```

Comment:
When using vagrant-bindfs, the convention appears to be...
1. do a regular vagrant folder sync from `somedir` to `/vagrant/somedir-nfs`
2. do a bindfs sync from `/vagrant/somedir-nfs` to `/final_path_on_vm/somedir`

Combined with the `File.basename()` adjustment to `nfs_path()` proposed in a separate comment, the following seems to handle it.
```diff
+ destination_folder = folder.fetch('bindfs', true) ? nfs_path(folder['destination']) : folder['destination']

- config.vm.synced_folder folder['local_path'], folder['destination'], options
+ config.vm.synced_folder folder['local_path'], destination_folder, options

  if folder.fetch('bindfs', true)
-   config.bindfs.bind_folder folder['local_path'], folder['destination'], options
+   config.bindfs.bind_folder destination_folder, folder['destination'], folder.fetch('bindfs_options', {})
```
The other piece to this is the `bindfs_options`, which are different from a regular vagrant sync dir `options` and must be treated differently. The above appears to successfully handle the following example 

```
# vagrant.local.yml
vagrant_synced_folders:
  - local_path: ../db
    destination: /home/vagrant/db
    create: false
    type: nfs
    bindfs: true
    bindfs_options:
      o: 'nonempty'
      p: '0644,a+D'
  - local_path: ../test
    destination: /home/vagrant/test
    create: false
    type: virtualbox
    bindfs: false
    mount_options:
      - 'dmode=755'
      - 'fmode=644'
```